backintime-library/runtime/src/nativeMain/kotlin/com/github/kitakkun/backintime/runtime/WeakReference.native.kt

```diff
@@ -0,0 +1,13 @@
+package com.github.kitakkun.backintime.runtime
+
+import kotlin.experimental.ExperimentalNativeApi
+
+@OptIn(ExperimentalNativeApi::class)
+class WeakReferenceImpl<T : Any>(value: T) : WeakReference<T> {
+    private val ref = kotlin.native.ref.WeakReference(value)
+    override fun get(): T? = ref.get()
+}
+
+actual fun <T : Any> weakReferenceOf(value: T): WeakReference<T> {
+    return WeakReferenceImpl(value)
+}
```

Comment:
Currently, the redefinition of class causes compilation errors. FYI: https://youtrack.jetbrains.com/issue/KT-63728/K2-KMP-exception-when-actualizing-same-expect-class-twice-in-different-source-sets